src/transaction/mod.rs

```diff
@@ -24,11 +26,42 @@ pub struct Timestamp {
     pub logical: i64,
 }
 
-pub enum Mutation {
-    Put(Key, Value),
-    Del(Key),
-    Lock(Key),
-    Rollback(Key),
+#[derive(Debug, Clone)]
+enum Mutation {
+    Put(Value),
+    Del,
+    Lock,
+}
+
+impl Mutation {
+    fn with_key(self, key: Key) -> kvrpcpb::Mutation {
```

Comment:
I would rename this `into_proto` or `into_proto_with_key` (`with_*` is the idiomatic name for a method which takes a function to operate on an internal value).